crates/re_sdk/src/spawn.rs

```diff
@@ -0,0 +1,166 @@
+/// Options to control the behavior of [`spawn`].
+///
+/// Refer to the field-level documentation for more information about each individual options.
+///
+/// The defaults are ok for most use cases: `SpawnOptions::default()`.
+/// Use the builder pattern to customize them further:
+/// ```no_run
+/// let opts = re_sdk::SpawnOptions::default().with_port(1234u16).with_memory_limit("25%");
+/// ```
+#[derive(Debug, Clone, Default)]
+pub struct SpawnOptions {
+    /// The port to listen on.
+    ///
+    /// Defaults to `9876` if unspecified.
+    pub port: Option<u16>,
+
+    /// An upper limit on how much memory the Rerun Viewer should use.
+    /// When this limit is reached, Rerun will drop the oldest data.
+    /// Example: `16GB` or `50%` (of system total).
+    ///
+    /// Defaults to `75%` if unspecified.
+    pub memory_limit: Option<String>,
+
+    /// Specifies the name of the Rerun executable.
+    ///
+    /// You can omit the `.exe` suffix on Windows.
+    ///
+    /// Defaults to `rerun` if unspecified.
+    pub executable_name: Option<String>,
```

Comment:
Why not use a `String` here, with `"rerun"` as the default?

Reply:
Sure